blockchain/types/transaction.go

```diff
@@ -292,6 +292,14 @@ func (tx *Transaction) EffectiveGasPrice(baseFee *big.Int) *big.Int {
 	return tx.GasPrice()
 }
 
+func (tx *Transaction) AccessList() AccessList {
+	if tx.Type() == TxTypeEthereumAccessList || tx.Type() == TxTypeEthereumDynamicFee {
```

Comment:
```suggestion
	if tx.IsEthTypedTransaction() {
```
How about using `IsEthTypedTransaction()`?

Reply:
I thought new ethereum tx types may not have AccessList.